src/test/java/ua/bookstore/online/service/impl/ShoppingCartServiceImplTest.java

```diff
@@ -0,0 +1,220 @@
+        when(cartMapper.toDto(any(ShoppingCart.class))).thenReturn(expected);
+
+        // When
+        ShoppingCartDto actual = shoppingCartService.getShoppingCartWithCartItems(user);
+
+        // Then
+        assertNotNull(actual);
+        assertTrue(actual.cartItems().isEmpty());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("Add cart item to existing shopping cart")
+    void addCartItem_AddToExistingCart_ReturnsCartItemResponseDto() {
+        // Given
+        CartItemRequestDto requestDto = getCartItemRequestDto();
+        User user = getUser();
+        ShoppingCart shoppingCart = getShoppingCart();
+        CartItemResponseDto expected = getCartItemResponseDto();
+
+        // Mocking behavior
+        when(cartRepository.findByUser(user)).thenReturn(Optional.of(shoppingCart));
+        when(cartItemService.add(requestDto, shoppingCart)).thenReturn(expected);
+
+        // When
+        CartItemResponseDto actual = shoppingCartService.addCartItem(requestDto, user);
+
+        // Then
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("Add cart item to non-existing shopping cart")
+    void addCartItem_AddToNonExistingCart_ReturnsCartItemResponseDto() {
+        // Given
+        CartItemRequestDto requestDto = getCartItemRequestDto();
+        User user = getUser();
+        ShoppingCart shoppingCart = getShoppingCart();
+        CartItemResponseDto expected = getCartItemResponseDto();
+
+        // Mocking behavior
+        when(cartRepository.findByUser(user)).thenReturn(Optional.empty());
+        when(cartRepository.save(any(ShoppingCart.class))).thenReturn(shoppingCart);
+        when(cartItemService.add(requestDto, shoppingCart)).thenReturn(expected);
+
+        // When
+        CartItemResponseDto actual = shoppingCartService.addCartItem(requestDto, user);
+
+        // Then
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("Update cart item in existing shopping cart")
+    void updateCartItem_UpdateItemInExistingCart_ReturnsQuantityDto() {
+        // Given
+        QuantityDto quantityDto = getQuantityDto();
+        User user = getUser();
+        ShoppingCart shoppingCart = getShoppingCart();
+
+        // Mocking behavior
+        when(cartRepository.findByUser(user)).thenReturn(Optional.of(shoppingCart));
+        when(cartItemService.changeQuantity(ID_1, quantityDto, shoppingCart))
+                .thenReturn(quantityDto);
+
+        // When
+        QuantityDto actual = shoppingCartService.updateCartItem(ID_1, quantityDto, user);
+
+        // Then
+        assertNotNull(actual);
+        assertEquals(quantityDto, actual);
+    }
+
+    @Test
+    @DisplayName("Update cart item in non-existing shopping cart")
+    void updateCartItem_UpdateItemInNonExistingCart_ReturnsQuantityDto() {
```

Comment:
another name incorrespondance to the body, please double-check other classes & methods as well